### PR TITLE
Fix setting collections in DebugDynamicConfigSource

### PR DESCRIPTION
**Pull request by chrisfearless**
Collection's toString wraps the values in square brackets, ex. "[a,b,c]" but the CSV parser doesn't expect the square brackets which prevents setting it via the DebugDynamicConfigSource unless you use the raw fireEvent. Options are to change the CSV regex in ConfigValueConverters to allow CSV wrapped in square brackets (which technically isn't valid csv) or to detect and convert collections to parsable CSV.

I chose the latter option since this really only affects setting the values dynamically in tests.

### CODE REVIEW

File: ice/src/main/java/com/kik/config/ice/source/DebugDynamicConfigSource.java [hunk 1]
```diff
@@ -103,13 +107,22 @@ protected DebugDynamicConfigSource(ConfigDescriptorHolder configDescriptorHolder
...
         {
             @Override
             public void toValue(final V value)
             {
-                final String stringValue = value == null ? "" : String.valueOf(value);
+                final String stringValue;
+
+                if (configClass != null && Collection.class.isAssignableFrom(configClass)) {
+                    final Collection collection = (Collection) value;
+                    stringValue = collection == null ? "" : (String) collection.stream().collect(Collectors.joining(","));
```
Comment (whited) [inline ice/src/main/java/com/kik/config/ice/source/DebugDynamicConfigSource.java]: Two potential problems with this:
(1) what if it's a Collection of something other than string? (eg: `List<Integer>`).  I think there should be a step before the `.collect` to explicitly coerce each value into string form.
(2) If any of the values contains a `,` it will mess up the CSV parsing.  There should be a map step that checks for `,` in the value, and escapes it.  Based on the parser logic, you would want to double-up any `"` and also enclose the whole value with `"`

Comment (chrisfearless) [reply ice/src/main/java/com/kik/config/ice/source/DebugDynamicConfigSource.java]: 1) it didn't appear like there were value converters for collections of things other than string (give guice bind errors), but I could see if I can just add one and test, or just future proof this
2) will look into it

**Comment by chrisfearless** [comment]
Made a change to preserve quotes by escaping and then wrapping in quotes so, although there may be edge cases, it should convert to a collection of exactly what you put in.
